src/Kestrel/Internal/ConfigurationReader.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal
+{
+    internal class ConfigurationReader
```

Comment:
Nit: you could consider just making this into two internal static extension methods on IConfiguration

`IConfiguration.ReadCertificates(this IConfiguration)`
`IConfiguration.ReadEndpoints(this IConfiguration)`

Reply:
No need when this is only for internal usage.